src/dialog-editor/components/toolbox/toolboxComponent.ts

```diff
@@ -0,0 +1,132 @@
+class DialogField {
+  public icon: string;
+  public label: string;
+  public placeholders: any;
+
+  constructor(type: string,
+              icon: string,
+              label: string,
+              options: any = {}) {
+    this.icon = icon;
+    this.label = label;
+    this.placeholders = Object.assign({
```

Comment:
I think [object-rest-spread](https://github.com/sebmarkbage/ecmascript-rest-spread) looks better, than `Object.assign` but that's on preferences.
```typescript
this.placeholders = {
      name: '',
      description: '',
      type: type,
      display: 'edit',
      display_method_options: {},
      read_only: false,
      required: false,
      required_method_options: {},
      default_value: '',
      values_method_options: {},
      label: label,
      position: 0,
      dynamic: false,
      show_refresh_button: false,
      load_values_on_init: false,
      auto_refresh: false,
      trigger_auto_refresh: false,
      reconfigurable: false,
      visible: true,
      options: {
        protected: false,
      },
      resource_action: {resource_type: 'DialogField', ae_attributes: {}},
      ...options
}
```